scripts/check_backend_associated_test_file_test.py

```diff
@@ -44,27 +46,31 @@ def mock_error(msg: str) -> None:
         self.print_swap = self.swap(builtins, 'print', mock_print)
         self.swap_logging = self.swap(logging, 'error', mock_error)
         self.swap_exit = self.swap(sys, 'exit', lambda _: None)
+
+    def test_checks_fail_when_a_backend_file_lacks_associated_test_file(
+            self) -> None:
         # Creating a backend file.
-        with open(self.backend_file, 'w', encoding='utf8') as f:
-            f.write('Example code')
+        temp_file1 = tempfile.NamedTemporaryFile()
+        setattr(temp_file1, 'name', self.backend_file)
```

Comment:
Can we do something like this instead? I don't think changing the name of a temporary file is standard practice (since that won't change the actual name of the file on the filesystem). I do recognize though that mypy doesn't understand that this file will have a name.

```suggestion
        self.backend_file = getattr(temp_file1, 'name')
```

Reply:
Hi @U8NWXD, we need this to be done because we differentiate temporary files into backend files, backend test files and frontend files on the basis of their names (like backend files end with `.py`).

Reply:
Ahh okay. In that case, can you create a temporary directory and then create your files (with whatever names you choose) there? That way you get control over the file names while also using tempfile (and without the hack of changing the `name` attribute of the file object)

Reply:
Done